src/Processors/Formats/Impl/ArrowFieldIndexUtil.h

```diff
@@ -227,16 +231,26 @@ class ArrowFieldIndexUtil
         }
         else if (const auto * type_array = typeid_cast<const DB::DataTypeArray *>(nested_type.get()))
         {
-            findRequiredIndices(name, transformed_name, header_index, type_array->getNestedType(), field_indices, added_indices, required_indices, file, clickhouse_to_parquet_names);
+            findRequiredIndices(Nested::concatenateName(name, "list.element"), Nested::concatenateName(transformed_name, "list.element"), header_index, type_array->getNestedType(), field_indices, added_indices, required_indices, file, clickhouse_to_parquet_names);
             return;
         }
         else if (const auto * type_map = typeid_cast<const DB::DataTypeMap *>(nested_type.get()))
         {
             findRequiredIndices(name, transformed_name, header_index, type_map->getKeyType(), field_indices, added_indices, required_indices, file, clickhouse_to_parquet_names);
-            findRequiredIndices(name, transformed_name, header_index, type_map->getValueType(), field_indices, added_indices, required_indices, file, clickhouse_to_parquet_names);
+            findRequiredIndices(Nested::concatenateName(name, "value"), Nested::concatenateName(transformed_name, "value"), header_index, type_map->getValueType(), field_indices, added_indices, required_indices, file, clickhouse_to_parquet_names);
             return;
         }
-        auto it = field_indices.find(transformed_name);
+        auto name_to_search = transformed_name;
+        auto it = field_indices.find(name_to_search);
+        if (it == field_indices.end())
+        {
+            /// There are some transforms with non equals input and output formats
+            /// For example geoparquet has binary as input format and 3d list in output format
+            /// In such situation we need to delete all list attributes from name to get correct representation
+            while (name_to_search.ends_with("list.element"))
+                name_to_search = name_to_search.substr(0, name_to_search.size() - 13);
```

Comment:
please make `"list.element" std::string_view static constant an use it everywhere, including this place